packages/aws-cdk-lib/core/lib/removal-policies.ts

```diff
@@ -0,0 +1,154 @@
+import { IConstruct } from 'constructs';
+import { Aspects, IAspect } from './aspect';
+import { CfnResource } from './cfn-resource';
+import { RemovalPolicy } from './removal-policy';
+
+/**
+ * Properties for applying a removal policy
+ */
+export interface RemovalPolicyProps {
+  /**
+   * Apply the removal policy only to specific resource types.
+   * Can be a CloudFormation resource type string (e.g., 'AWS::S3::Bucket').
+   * @default - apply to all resources
+   */
+  readonly applyToResourceTypes?: string[];
+
+  /**
+   * Exclude specific resource types from the removal policy.
+   * Can be a CloudFormation resource type string (e.g., 'AWS::S3::Bucket').
+   * @default - no exclusions
+   */
+  readonly excludeResourceTypes?: string[];
```

Comment:
It would be good to explain the behavior when the same type is specified on both `applyToResourceTypes` and `excludeResourceTypes`.